### PR TITLE
Added code to vet the poller input images viability for drizzling

### PR DESCRIPTION
**Pull request by mdlpstsci**
This PR addresses Issue #517.  Additional changes are pending feedback from individuals in operations testing.

### CODE REVIEW

File: drizzlepac/hapsequencer.py [hunk 1]
```diff
@@ -404,8 +404,9 @@ def run_hap_processing(input_filename, diagnostic_mode=False, use_defaults_confi
         log.info("Return condition {}".format(return_value))
         logging.shutdown()
         # Append total trailer file (from astrodrizzle) to all total log files
-        for tot_obj in total_list:
-            proc_utils.append_trl_file(tot_obj.trl_filename, logname, clean=False)
+        if product_list:
```
Comment (stsci-hack) [inline drizzlepac/hapsequencer.py]: The code being turned on/off here only depend on the total_list, so it is confusing to trigger on the product_list instead.

File: drizzlepac/hlautils/analyze.py [hunk 2]
```diff
@@ -43,6 +45,43 @@ class Messages(Enum):
...
     OK, WARN, NOPROC = 1, -1, -2
 
+
+def analyze_wrapper(input_file_list):
+    """
+    Thin wrapper for the analyze_data function to return a list of viable images.
+
+    Parameters
+    ==========
+    input_file_list: list
```
Comment (stsci-hack) [inline drizzlepac/hlautils/analyze.py]:
```suggestion
    input_file_list : list
```
This simple change will allow Sphinx to render the parameter correctly.

File: drizzlepac/hlautils/analyze.py [hunk 3]
```diff
@@ -43,6 +45,43 @@ class Messages(Enum):
...
+        List containing FLT and/or FLC filenames for all input images which comprise an associated
+        dataset where 'associated dataset' may be a single image, multiple images, an HST
+        association, or a number of HST associations
+
+    Returns
+    =======
+    viable_images_list: list
```
Comment (stsci-hack) [inline drizzlepac/hlautils/analyze.py]:
```suggestion
    viable_images_list : list
```

File: drizzlepac/hapsequencer.py (hunk 1 above)
Comment (mdlpstsci) [reply drizzlepac/hapsequencer.py]: Fixed.

File: drizzlepac/hlautils/analyze.py (hunk 2 above)
Comment (mdlpstsci) [reply drizzlepac/hlautils/analyze.py]: Fixed.

File: drizzlepac/hlautils/analyze.py (hunk 3 above)
Comment (mdlpstsci) [reply drizzlepac/hlautils/analyze.py]: Fixed.

**Pull request merged**